app/src/shared/hooks/useAccountAddress.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import { useState, useEffect } from 'react'
+import { getWeb3 } from '$shared/web3/web3Provider'
+import Web3Poller from '$shared/web3/web3Poller'
+import useIsMounted from '$shared/hooks/useIsMounted'
+
+export default (): ?string => {
+    const [address, setAddress] = useState()
+
+    const isMounted = useIsMounted()
+
+    useEffect(() => {
+        (async () => {
+            try {
+                return await getWeb3().getDefaultAccount()
+            } catch (e) {
+                return undefined
+            }
+        })().then((addr) => {
+            if (isMounted()) {
+                setAddress(addr)
+            }
```

Comment:
I'd prefer this form because it's a bit easier to read:

```javascript
const loadAddress = async () => {
    try {
        const address = await getWeb3().getDefaultAccount()
        if (address && isMounted()) {
            setAddress(address)
        }
    }
    catch (e) {
        setAddress(undefined)
    }
}
loadAddress()
```

Reply:
I agree, looks more readable. This is "borrowed" from #832 which is not yet merged.